### ISSUE TITLE
Calling `generateFunctionCode` twice results in runtime exception

### ISSUE THREAD

**Issue by manuranga** [ballerina-platform/nballerina#1164]
Steps to reproduce:
In `compiler/modules/nback/module.bal` and an additional call to `generateFunctionCode`

Failing tests:
`testCompileVPO#'09-exact/record1-p.bal'`, `testCompileVPO#'09-exact/push2-v.bal'`, `testCompileVPO#'09-exact/push1-p.bal'`

Stack trace:
```
                    recursive list atom not defined for: 1

                        at wso2.nballerina.types.0.Env:getRecListAtomType(core.bal:179)
                        wso2.nballerina.types.0.Context:listAtomType(core.bal:270)
                        wso2.nballerina.types.0:listFormulaIsEmpty(list.bal:161)
                        wso2.nballerina.types.0:listFormulaIsEmpty$lambda6$(list.bal:148)
                        wso2.nballerina.types.0:bddEvery(common.bal:92)
                        wso2.nballerina.types.0:bddEvery(common.bal:95)
                        wso2.nballerina.types.0:listBddIsEmpty(list.bal:148)
                        wso2.nballerina.types.0:listBddIsEmpty$lambda4$(list.bal:140)
                        wso2.nballerina.types.0:memoSubtypeIsEmpty(common.bal:60)
                        wso2.nballerina.types.0:listSubtypeIsEmpty(list.bal:140)
                        wso2.nballerina.types.0:isEmpty(core.bal:836)
                        wso2.nballerina.front.0:resolveSubsetTypeDesc(resolveTypes.bal:101)
                        wso2.nballerina.front.0.StmtContext:resolveTypeDesc(stmt.bal:212)
                        wso2.nballerina.front.0:codeGenVarDeclStmt(stmt.bal:822)
                        wso2.nballerina.front.0:codeGenStmt(stmt.bal:339)
                        wso2.nballerina.front.0:codeGenScope(stmt.bal:274)
                        wso2.nballerina.front.0:codeGenFunction(stmt.bal:226)
                        wso2.nballerina:generateFunctionCode(front.bal:48)
                        wso2.nballerina.nback.0:buildModule(module.bal:59)
                        wso2.nballerina.0.LlvmEmitter:emitModule(compile.bal:43)
                        wso2.nballerina.0:processModule(compile.bal:123)
                        wso2.nballerina.0:compileBalFile(compile.bal:108)
                        wso2.nballerina$test.0.tests.testSuite:testCompileFile(tests/testSuite.bal:198)
                        wso2.nballerina$test.0.tests.testSuite:testCompileVPO(tests/testSuite.bal:22)
```

**Comment by manuranga** [comment]
This is a regression issue due to #1127

**Comment by manuranga** [comment]
Part of the fix has been accidentally removed during #1162
